11_FFTBloom/main.cpp

```diff
@@ -0,0 +1,1192 @@
+
+			// Give them debug names
+			m_srcImageView->setObjectDebugName("Source image view");
+			m_srcImageView->getCreationParameters().image->setObjectDebugName("Source Image");
+			m_kerImageView->setObjectDebugName("Bloom kernel image view");
+			m_kerImageView->getCreationParameters().image->setObjectDebugName("Bloom kernel Image");
+
+			// The down-cast should not fail!
+			assert(m_srcImageView);
+			assert(m_kerImageView);
+
+			// Required size for uploads
+			auto srcImageDims = m_srcImageView->getCreationParameters().image->getCreationParameters().extent;
+			auto kerImageDims = m_kerImageView->getCreationParameters().image->getCreationParameters().extent;
+			// Add a bit extra because EXR has alpha
+			uint32_t srcImageSize = srcImageDims.height * srcImageDims.width * srcImageDims.depth * (CHANNELS + 1) * sizeof(float32_t);
+			uint32_t kerImageSize = kerImageDims.height * kerImageDims.width * kerImageDims.depth * (CHANNELS + 1) * sizeof(float32_t);
+
+			m_utils = make_smart_refctd_ptr<IUtilities>(smart_refctd_ptr(m_device), smart_refctd_ptr(m_logger), srcImageSize, srcImageSize + kerImageSize);
+
+			// Now convert uploads
+			// Get graphics queue for image transfer
+			auto graphicsQueue = getQueue(IQueue::FAMILY_FLAGS::GRAPHICS_BIT);
+			m_intendedSubmit.queue = graphicsQueue;
+			// Set up submit for image transfers
+			// wait for nothing before upload
+			m_intendedSubmit.waitSemaphores = {};
+			m_intendedSubmit.prevCommandBuffers = {};
+			// fill later
+			m_intendedSubmit.scratchCommandBuffers = {};
+			m_intendedSubmit.scratchSemaphore = {
+				.semaphore = m_scratchSemaphore.get(),
+				.value = 0,
+				// because of layout transitions
+				.stageMask = PIPELINE_STAGE_FLAGS::ALL_COMMANDS_BITS
+			};
+
+			// Create a command buffer for graphics submit, though it has to be resettable
+			smart_refctd_ptr<IGPUCommandBuffer> graphicsCmdBuf;
+			smart_refctd_ptr<nbl::video::IGPUCommandPool> cmdpool = m_device->createCommandPool(graphicsQueue->getFamilyIndex(), IGPUCommandPool::CREATE_FLAGS::RESET_COMMAND_BUFFER_BIT);
+			if (!cmdpool->createCommandBuffers(IGPUCommandPool::BUFFER_LEVEL::PRIMARY, 1u, &graphicsCmdBuf))
+				return logFail("Failed to create Command Buffers!\n");
+
+			// Needs to be open for utilities
+			graphicsCmdBuf->begin(IGPUCommandBuffer::USAGE::ONE_TIME_SUBMIT_BIT);
+
+			IQueue::SSubmitInfo::SCommandBufferInfo graphicsCmdbufInfo = { graphicsCmdBuf.get() };
+			m_intendedSubmit.scratchCommandBuffers = { &graphicsCmdbufInfo,1 };
+
+			// We need compute queue to be the owner of the images after transfer + layout transition
+			struct SConvertParams : CAssetConverter::SConvertParams
+			{
+				virtual inline uint32_t getFinalOwnerQueueFamily(const IGPUImage* image, const core::blake3_hash_t& createdFrom, const uint8_t mipLevel)
+				{
+					return computeFamilyIndex;
+				}
+
+				uint32_t computeFamilyIndex;
+			};
+			SConvertParams params = {};
+			params.computeFamilyIndex = queueFamilyIndex;
```

Comment:
don't override this, if you leave as `FamilyIngored` then the owner will stay the queue you gave in the `m_intendedSubmit`